src/test/kotlin/service/user/DeleteUserTest.kt

```diff
@@ -0,0 +1,84 @@
+package se.uulm.snowballr.backend.service.user
+
+import io.mockk.coEvery
+import kotlinx.coroutines.DelicateCoroutinesApi
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertDoesNotThrow
+import org.junit.jupiter.api.assertThrows
+import se.uulm.snowballr.backend.DataBuilder
+import se.uulm.snowballr.backend.db.dummyUserId
+import se.uulm.snowballr.backend.model.SnowballRException
+import se.uulm.snowballr.backend.model.SnowballRException.UnauthorizedException
+import se.uulm.snowballr.backend.service.MainServiceTest
+import se.uulm.snowballr.backend.testCoroutine
+import snowballr.Base
+import snowballr.UserOuterClass.UserRole
+
+@ExperimentalCoroutinesApi
+@DelicateCoroutinesApi
+internal class DeleteUserTest : MainServiceTest() {
+    @Test
+    fun `When an user that is not an admin tries to delete another user, then an unauthorized exception is thrown`() =
+        testCoroutine {
+            val firstUser = DataBuilder.createExampleUser(role = UserRole.USER_ROLE_DEFAULT)
+            val request =
+                Base.Id
+                    .newBuilder()
+                    .setId(firstUser.id.toString())
+                    .build()
+            val secondUser = DataBuilder.createExampleUser(role = UserRole.USER_ROLE_DEFAULT)
+
+            coEvery { userRepoMock.getUserById(dummyUserId!!) } returns secondUser
+
+            assertThrows<UnauthorizedException.All.User> { mainService.softDeleteUser(request) }
+        }
+
+    @Test
+    fun `When an user tries to delete another admin user, then an failed precondition exception is thrown`() =
```

Comment:
This makes it more clear.
```suggestion
    fun `When an admin tries to delete another admin user, then an failed precondition exception is thrown`() =
```